src/test/java/org/assertj/core/api/map/MapAssert_containsKey_Test.java

```diff
@@ -35,4 +44,25 @@
   protected void verify_internal_effects() {
     verify(maps).assertContainsKeys(getInfo(assertions), getActual(assertions), array("key1"));
   }
+
+  @ParameterizedTest
+  @MethodSource("maps_without_null_keys_or_values")
+  protected void should_fail_when_asserting_for_null_on_map_that_does_not_contain_null(final Map<?, ?> actual) {
```

Comment:
With the current architecture, the best would be adding cases to the corresponding test class in `org.assertj.core.internal.maps`.